sympy/printing/pycode.py

```diff
@@ -526,6 +526,56 @@ def _print_DotProduct(self, expr):
                                self._print(arg1),
                                self._print(arg2))
 
+    def _print_ZeroMatrix(self, expr):
+        return '{}({})'.format(self._module_format('numpy.zeros'),
+            self._print(expr.shape))
+
+    def _print_OneMatrix(self, expr):
+        return '{}({})'.format(self._module_format('numpy.ones'),
+            self._print(expr.shape))
+
+    def _print_FunctionMatrix(self, expr):
+        from sympy.core.function import Lambda
+        from sympy.abc import i, j
+        lamda = expr.lamda
+        if not isinstance(lamda, Lambda):
+            lamda = Lambda((i, j), lamda(i, j))
+        return '{}(lambda {}: {}, {})'.format(self._module_format('numpy.fromfunction'),
+            ', '.join(self._print(arg) for arg in lamda.args[0]),
+            self._print(lamda.args[1]), self._print(expr.shape))
+
+    def _print_HadamardProduct(self, expr):
+        func = self._module_format('numpy.multiply')
+        return ''.join('{}({}, '.format(func, self._print(arg)) \
+            for arg in expr.args[:-1]) + "{}{}".format(self._print(expr.args[-1]),
+            ')' * (len(expr.args) - 1))
+
+    def _print_KroneckerProduct(self, expr):
+        func = self._module_format('numpy.kron')
+        return ''.join('{}({}, '.format(func, self._print(arg)) \
+            for arg in expr.args[:-1]) + "{}{}".format(self._print(expr.args[-1]),
+            ')' * (len(expr.args) - 1))
+
+    def _print_Adjoint(self, expr):
+        return '{}.getH()'.format(self._print(expr.arg))
+
+    def _print_DiagonalOf(self, expr):
+        return '{}({})'.format(self._module_format('numpy.diag'), self._print(expr.arg))
+
+    def _print_DiagonalizeVector(self, expr):
+        return '{}({})'.format(self._module_format('numpy.diagflat'),
+            self._print(expr.args[0]))
+
+    def _print_DiagonalMatrix(self, expr):
+        zero = 'temp = {}({})'.format(self._module_format('numpy.zeros'),
+            self._print(expr.shape))
+        # will not work if both dimensions of expr are symbolic
+        diag_length = expr.diagonal_length
+        fill_diag = 'temp[{}({})] = {}'.format(
+            self._module_format('numpy.diag_indices'), diag_length,
+            self._print_DiagonalOf(expr))
+        return '{}\n{}'.format(zero, fill_diag)
```

Comment:
It's preferred that the code printers print a single expression, but it looks like this prints two separate statements. Maybe we can do something with `numpy.eye` (like `numpy.multiply(X, numpy.eye(n, m))`).

Reply:
I am assuming you mean ` numpy.matmul(X, numpy.eye(n, m)) `. `numpy.multiply() ` does element-wise multiplication. Yeah, that should work. I have no idea why I didn't think of that!

I guess, I will replace the string with -
 ` numpy.matmul(numpy.diagflat(numpy.diag(X)), numpy.eye(n, m)) `

Reply:
Element wise multiplication should zero out everything not on the diagonal

Reply:
Ah, I misinterpreted what you meant by ` X `. I get what you mean, yeah.